src/markov/ddp.jl

```diff
@@ -501,6 +505,40 @@ function solve(ddp::DiscreteDP{T}, v_init::AbstractVector{T},
     ddpr
 end
 
+"""
+Solve a finite horizon discrete dynamic program by backward induction with
+stationary reward and transition probability functions and discount factor.
+
+##### Parameters
+
+- `ddp::DiscreteDP{T}` : Object that contains the Model Parameters
+- `J::Integer`: Number of decision periods
+- `v_term::AbstractVector{T}=zeros(num_states(ddp))`: Terminal value function,
+  of length equal to n (the number of states)  
+
+##### Returns
+
+- `vs::Array{Float64,2}`: Matrix that contains the optimal value function 
+  at each period
+- `sigmas::Array{Int,2}`: Matrix that contains the optimal policy function
+  at each period
+"""
+function backward_induction(ddp::DiscreteDP{T}, J::Integer,
+                            v_term::AbstractVector{T}=
+                            zeros(num_states(ddp))) where {T}
+    n = num_states(ddp)
+    vs = zeros(J+1,n)
+    vs[end,:] = v_term
+    v_temp = v_term
+    sigmas = zeros(Int, J, n)
+    sigma_temp = sigmas[end,:]
+    @inbounds for j in J+1: -1: 2
+        v_temp, sigma_temp = bellman_operator!(ddp, v_temp, sigma_temp)
+        vs[j-1,:], sigmas[j-1,:] = v_temp, sigma_temp
+    end
```

Comment:
Is it possible to avoid allocating the temporary arrays?

Reply:
We can rewrite this as 
```julia
@inbounds for j in J+1: -1: 2
        vs[j-1, :], sigmas[j-1,:] =  @views bellman_operator!(ddp,vs[j,:],sigmas[j-1,:])
    end
```
Thanks. I'll fix it.

Reply:
Here, we can rewrite this as 
```julia
@inbounds for j in J+1: -1: 2
    @views bellman_operator!(ddp,vs[j,:],vs[j-1,:],sigmas[j-1,:])
end
```

Thanks! I'll fix it.

Reply:
By the way, Julia being column-major, the values at period j should be stored the j-th *column* (`vs` and `sigmas` should be of shape `(n, J+1)` and `(n, J)`).

Reply:
For internal consistency, I adopt the above written style, but I follow your suggestion.